src/poweredup-browser.ts

```diff
@@ -132,31 +132,29 @@ export class PoweredUP extends EventEmitter {
             let buf: Buffer = Buffer.alloc(0);
             device.subscribeToCharacteristic(Consts.BLECharacteristic.LPF2_ALL, (data: Buffer) => {
                 buf = Buffer.concat([buf, data]);
-                const len = buf[0];
-                if (len >= buf.length) {
```

Comment:
I may be wrong but this condition does not look good to me: expected length greater than buffer size ?

Reply:
BLE has QoS to scale MTU size depending on network conditions. It's rare, but under certain poor/highly contested conditions the sender may not be able to fit the whole message into a single packet, meaning we get partial data instead of a full message.

The first byte of the message we receive tells us the length of data we should expect. If we don't have all the data yet, we skip parsing, concatenating data until we get a full message (line 134).

Reply:
> I think the length check needs to be reintroduced, but lgtm otherwise. :) I need to spend a bit of time testing to ensure the removal of the `process.nextTick` doesn't cause any issues.

The length check is still there but in a while to process as many messages as possible.
My comment was about the sign of the comparison in the if: it seems to parse messages if the buffer is equal or **smaller** than the advertised message length.

I will do some more tests tomorrow to check with the boost hub.

Reply:
lawl, good catch - the comparison is turned around. Turns out I shouldn't be allowed anywhere near a text editor. :D